src/Http/Http.Abstractions/test/MapPathMiddlewareTests.cs

```diff
@@ -39,8 +39,8 @@ private static void UseNotImplemented(IApplicationBuilder app)
         [Fact]
         public void NullArguments_ArgumentNullException()
         {
-            var builder = new ApplicationBuilder(serviceProvider: null);
-            var noMiddleware = new ApplicationBuilder(serviceProvider: null).Build();
+            var builder = new ApplicationBuilder(serviceProvider: null!);
+            var noMiddleware = new ApplicationBuilder(serviceProvider: null!).Build();
```

Comment:
Suggestion only: Seem to be working around the nullability of this constructor parameter a fair amount. Might be easier to either provide a default constructor or allow `null` in this constructor for testing purposes. Either way, shouldn't need to change the type of the `ApplicationServices` property (though it's odd the `ApplicationBuilder` properties aren't nullable given `GetProperty(...)` can return `null`).

Reply:
@Tratcher, what do you think? https://github.com/dotnet/aspnetcore/blob/master/src/Http/Http/src/Builder/ApplicationBuilder.cs#L22

Reply:
I guess I can do that in a follow up. Let's get this PR in

Reply:
The API is very inconsistent.

1. Adding a parameterless constructor would simplify usage in tests.
2. ApplicationServices_Get should return a static no-op instance rather than null.
3. ServerFeatures should also return a read-only no-op instance.

Reply:
Captured this in a separate feedback. https://github.com/dotnet/aspnetcore/issues/23232.